resources/get-customers.ts

```diff
@@ -26,8 +27,27 @@ export function getCustomers(): pulumi.Output<PortalCustomer[]> {
       apiVersion: '2023-04-18',
     });
 
-    const result = await client.fetch("*[_type == 'customer']");
-    return z.array(portalCustomer).parse(result);
+    const result = await client.fetch<unknown[]>("*[_type == 'customer']");
+    const customers = result
+      .map(rawCustomer => {
+        const customer = portalCustomer.safeParse(rawCustomer);
+        if (!customer.success) {
+          const { id } = rawCustomer as { id: string };
+          pulumi.log.warn(
+            `Customer could not be added due to data not adhering to correct data structure. (${
+              id ?? 'Unknown ID'
+            })`,
+          );
```

Comment:
Since we are getting this error, does that mean something has been set incorrectly in sanity?

Reply:
Yes, which is fine for now IMHO.